framework/src/modules/poa/utils.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { utils } from '@liskhq/lisk-cryptography';
+import { ValidatorWeight } from './types';
+
+// Same as pos/utils/shuffleValidatorList
+export const shuffleValidatorList = (
+	previousRoundSeed1: Buffer,
+	addresses: ValidatorWeight[],
+): ValidatorWeight[] => {
+	const validatorList = [...addresses].map(validator => ({
+		...validator,
+	})) as { address: Buffer; roundHash: Buffer; weight: bigint }[];
+
+	for (const validator of validatorList) {
+		const seedSource = Buffer.concat([previousRoundSeed1, validator.address]);
+		validator.roundHash = utils.hash(seedSource);
+	}
+
+	validatorList.sort((validator1, validator2) => {
+		const diff = validator1.roundHash.compare(validator2.roundHash);
+		if (diff !== 0) {
+			return diff;
+		}
+
+		return validator1.address.compare(validator2.address);
+	});
+
+	return validatorList;
```

Comment:
Maybe I don't completely understand the TS language features, but isn't `validatorList` an array of `{Buffer, Buffer, bigint}` instead of `{Buffer, bigint}`? Will it comply the function signature `: ValidatorWeight[]`?

Reply:
yes, the function return signature is misleading as it also returns roundHash with each item. I have created a new type `ValidatorWeightWithRandomHash` to make it more clearer